mist-api/src/routes/genomes/get.js

```diff
@@ -22,17 +23,54 @@ module.exports = function(app, middlewares, routeMiddlewares) {
 				'taxonomy_id',
 			],
 		}),
-		// Provide for searching against name
+		// Provide for searching against name, taxonomy levels and assembly_level
+		//In a query all the terms except 'search' and 'assembly_level' are expected to be one word terms, nevetherless as a precaution
+		//all the terms are being taken and concatenated.
 		(req, res, next) => {
 			if (Reflect.has(req.query, 'search')) {
-				const searchTerms = util.splitIntoTerms(req.query.search)
-					.map((term) => `%${term}%`)
-				if (searchTerms.length > 0)
-					_.set(res.locals, 'criteria.where.name.$iLike.$any', searchTerms)
+				const searchTerm = util.splitIntoTerms(req.query.search)
+					.map((term) => `'(${term}:*)'`).join(" & ").replace(/' & '/g, ' & ')
+				if (searchTerm.length > 0)
+					_.set(res.locals, 'criteria.where.search', searchTerm)
```

Comment:
`search` is not a real column in the database. Rather we use the query parameter `search` as a generic source for finding whatever relevant matches we think best.